impeller/renderer/backend/gles/texture_gles.cc

```diff
@@ -130,6 +130,11 @@ struct TexImage2DData {
         external_format = GL_ALPHA;
         type = GL_UNSIGNED_BYTE;
         break;
+      case PixelFormat::kR8UNormInt:
```

Comment:
GL_RED is ES 3.0 and above only. A8 was chosen based on the ES 2.0 floors cap.

I agree that we should NOT depend on swizzles however. We observed some pretty dire performance cliffs with swizzles on older drivers. Though, to perform a swizzle via GL_TEXTURE_SWIZZLE_<foo> requires ES 3.0 too. So we can't do it on ES 2.0 (and we shouldn't).

I suggest having a `kGrayBitmap` that maps to the backend native format and have an ifdef in the shader to pick the right format directly.

Reply:
ack, will do some more digging on that. I thought I observed Skia consistently using red bitmaps but that might just be on newer devices.

Reply:
👍🏽  go/es2-reference and go/es3-reference might come in handy. I also [have Dash docsets for the different versions so checking up flag compatibility is easier](https://github.com/chinmaygarde/OpenGLESDashDocset). If you use Dash though.